src/main/java/io/gravitee/common/utils/RxHelper.java

```diff
@@ -336,15 +355,20 @@ private static <R extends Throwable> FlowableTransformer<R, R> takeThenThrow(fin
                         return Single.error(throwable);
                     }
                 })
-                .zipWith(Flowable.range(1, Integer.MAX_VALUE), (throwable, attemptNumber) -> attemptNumber)
-                .map(attemptNumber -> {
-                    long delayMs = Math.round(Math.pow(factor, attemptNumber.doubleValue() - 1) * timeUnit.toMillis(initialDelay));
-                    if (maxDelay != -1) {
-                        long maxDelayMs = timeUnit.toMillis(maxDelay);
-                        delayMs = Math.min(maxDelayMs, delayMs);
-                    }
-                    return delayMs;
-                })
+                .zipWith(Flowable.range(1, Integer.MAX_VALUE), Attempt::new)
+                .concatMapSingle(attempt ->
```

Comment:
Could you explain why you need to use a single here? I feel like you could stick with a simple map. WDYT ?